src/dns_internal.cc

```diff
@@ -1334,6 +1327,7 @@ idnsRead(int fd, void *)
     while (max) {
         --max;
         len = comm_udp_recvfrom(fd, rbuf, SQUID_UDP_SO_RCVBUF, 0, bugbypass);
+        fd_table[fd].bytesRead(len);
```

Comment:
Passing a negative `len` value to bytesRead(size_t) will create havoc. Please move this new line to where the old fd_bytes() call was. Please double check for similar bugs in the PR -- I have not.

Reply:
No. The bytesRead() implementation explicitly handles negative values for exactly this type of case.

Reply:
fixed in [bb0ee01](https://github.com/squid-cache/squid/pull/1352/commits/bb0ee01abadf8c307baaff54043962e528ef2c2e)